app/models/product_test.rb

```diff
@@ -108,10 +108,9 @@ def sample_and_duplicate_records(recs, ids, random: Random.new)
     dups = records.find(ids)
 
     recs, dups = randomize_clinical_data(recs, dups, random)
-
-    (random.rand(3) + 1).times do
+    dups.sample(random.rand(3) + 1, random: random).each do |rec|
```

Comment:
What we want here is `1`, `2`, or `3`. The way to do that is `random.rand(1..3)`.

As a side note, `random.rand(3) + 1` will also give you `1`, `2`, or `3`. `random.rand(1..4)` will include `4` as well.